interpretation/deepseismic_interpretation/azureml_pipelines/base_pipeline.py

```diff
@@ -0,0 +1,387 @@
+
+        :returns: parallel_run_config
+        :rtype: ParallelRunConfig
+        """
+        return ParallelRunConfig(source_directory=step['source_directory'],
+                                 entry_script=step['script'],
+                                 mini_batch_size=str(step['mini_batch_size']),
+                                 error_threshold=10,
+                                 output_action="summary_only",
+                                 environment=self._setup_env(step),
+                                 compute_target=self.compute_target,
+                                 node_count=step.get('node_count', 1),
+                                 process_count_per_node=step.get('processes_per_node', 1),
+                                 run_invocation_timeout=60)
+
+    def _create_pipeline_step(self, step, arguments, input_data, output=None, run_config=None):
+        """
+        function to create an AzureML pipeline step and apend it to the list of
+        steps that will make up the pipeline.
+
+        :param dict step: [required] dictionary containing the config parameters for this step.
+        :param list arguments: [required] list of arguments to be passed to the step.
+        :param DataReference input_data: [required] the input_data in AzureML for this step.
+        :param DataReference output: [required] output location in AzureML
+        :param ParallelRunConfig run_config: [optional] the run configuration for a MpiStep
+        """
+
+        if step['type'] == 'PythonScriptStep':
+            run_config = self._generate_run_config(step)
+            pipeline_step = PythonScriptStep(script_name=step['script'],
+                                             arguments=arguments,
+                                             inputs=[input_data],
+                                             outputs=output,
+                                             name=step['name'],
+                                             compute_target=self.compute_target,
+                                             source_directory=step['source_directory'],
+                                             allow_reuse=True,
+                                             runconfig=run_config)
+
+        elif step['type'] == 'MpiStep':
+            pipeline_step = MpiStep(name=step['name'],
+                                    source_directory=step['source_directory'],
+                                    arguments=arguments,
+                                    inputs=[input_data],
+                                    node_count=step.get("node_count", 1),
+                                    process_count_per_node=step.get("processes_per_node", 1),
+                                    compute_target=self.compute_target,
+                                    script_name=step['script'],
+                                    environment_definition=self._setup_env(step))
+
+        elif step['type'] == 'ParallelRunStep':
```

Comment:
Are we using this in the pipeline? Doesn't hurt to leave here, but just wondering

Reply:
Not currently, we do not have functionality for all of the steps @annazietlow initially included them so they could be worked on in the future. It might just be confusing to have in place. Would you prefer to remove these? I have a similar ask of the team in another conversation on this PR.

Reply:
They do work but they are just not used in the pipeline config that we have set up right now